addons/mail/controllers/discuss.py

```diff
@@ -303,6 +306,23 @@ def mail_message_remove_reaction(self, message_id, content):
     # Channel API
     # --------------------------------------------------------------------------
 
+    @http.route('/mail/channel/add_guest_as_member', methods=['POST'], type='json', auth='public')
+    def mail_channel_add_guest_as_member(self, channel_id, channel_uuid, **kwargs):
+        channel_sudo = request.env['mail.channel'].browse(channel_id).sudo().exists()
+        if not channel_sudo or not channel_sudo.uuid or not consteq(channel_sudo.uuid, channel_uuid):
+            raise NotFound()
+        if channel_sudo.channel_type == 'chat':
+            raise NotFound()
+        guest = channel_sudo.env['mail.guest']._get_guest_from_request(request)
+        # Only guests should take this route.
+        if not guest:
+            raise NotFound()
+        channel_partner = channel_sudo.env['mail.channel.partner']._get_as_sudo_from_request(request=request, channel_id=channel_id)
+        # Do not add the guest to channel members if they are already member.
+        if channel_partner:
+            raise UserError(_("The current guest is already member of the channel."))
```

Comment:
As the goal of the method is to have the guest be a member, I think we should avoid raising if he already is member, and just do nothing instead. No need to generate potentially problematic concurrency issues when the end result would be the same.